src/commerce-handler.js

```diff
@@ -1,66 +1,70 @@
+var eventMapping = {
+    10: 'ADD_TO_CART',
+    11: 'ProductRemoveFromCart',
+    12: 'INITIATE_PURCHASE',
+    13: 'INITIATE_PURCHASE',
+    14: 'CLICK_AD',
+    15: 'VIEW_ITEM',
+    16: 'PURCHASE',
+    17: 'ProductRefund',
+    18: 'VIEW_AD',
+    19: 'CLICK_AD',
+    20: 'ADD_TO_WISHLIST',
+    21: 'ProductRemoveFromWishlist',
+    22: 'VIEW_ITEM'
+}
+
+var aliasMapping = {
+    13: 'ProductCheckoutOption',
+    19: 'PromotionClick'
+}
+
 function CommerceHandler(common) {
     this.common = common || {};
 }
 
 CommerceHandler.prototype.logCommerceEvent = function(event) {
-    /*
-        Sample ecommerce event schema:
-        {
-            CurrencyCode: 'USD',
-            DeviceId:'a80eea1c-57f5-4f84-815e-06fe971b6ef2', // MP generated
-            EventAttributes: { key1: 'value1', key2: 'value2' },
-            EventType: 16,
-            EventCategory: 10, // (This is an add product to cart event, see below for additional ecommerce EventCategories)
-            EventName: "eCommerce - AddToCart",
-            MPID: "8278431810143183490",
-            ProductAction: {
-                Affiliation: 'aff1',
-                CouponCode: 'coupon',
-                ProductActionType: 7,
-                ProductList: [
-                    {
-                        Attributes: { prodKey1: 'prodValue1', prodKey2: 'prodValue2' },
-                        Brand: 'Apple',
-                        Category: 'phones',
-                        CouponCode: 'coupon1',
-                        Name: 'iPhone',
-                        Price: '600',
-                        Quantity: 2,
-                        Sku: "SKU123",
-                        TotalAmount: 1200,
-                        Variant: '64GB'
-                    }
-                ],
-                TransactionId: "tid1",
-                ShippingAmount: 10,
-                TaxAmount: 5,
-                TotalAmount: 1215,
-            },
-            UserAttributes: { userKey1: 'userValue1', userKey2: 'userValue2' }
-            UserIdentities: [
-                {
-                    Identity: 'test@gmail.com', Type: 7
-                }
-            ]
-        }
+    var event_data_and_custom_data = {
+        mpid: event.MPID,
+        affiliation: event.ProductAction.Affiliation,
+        coupon: event.ProductAction.CouponCode,
+        transaction_id: event.ProductAction.TransactionId,
+        shipping: event.ProductAction.ShippingAmount,
+        tax: event.ProductAction.TaxAmount,
+        revenue: event.ProductAction.TotalAmount,
+        ...event.EventAttributes,
+        ...event.UserAttributes
+    };
+
+    !!event.CurrencyCode ? (event_data_and_custom_data["currency"] = event.CurrencyCode) : console.log("");
```

Comment:
It's a bit funky but the intent is to avoid sending currency: null (or any other non-valid value) because annoyingly that'll cause us to reject the event entirely. Maybe there's a better way?

Reply:
got it.  in that case, no need for ternary still, and it should be the following:

```
if (event.CurrencyCode) {
    event_data_and_custom_data["currency"] = event.CurrencyCode;
}
```